docs/FeatureManagement/FeatureFlagv2.0.0.schema.json

```diff
@@ -0,0 +1,354 @@
+                "$id": "#/properties/allocation/properties/percentile",
+                "type": "array",
+                "title": "Percentile Allocation Collection",
+                "description": "A list of objects, each containing a variant name and percentage range for which that variant should be used.",
+                "items": {
+                    "$id": "#/properties/allocation/properties/percentile/items",
+                    "type": "object",
+                    "title": "Percentile Allocation",
+                    "required": [
+                        "variant",
+                        "from",
+                        "to"
+                    ],
+                    "properties": {
+                        "variant": {
+                            "$id": "#/properties/allocation/properties/percentile/items/properties/variant",
+                            "type": "string",
+                            "title": "Percentile Allocation Variant",
+                            "description": "The name of the variant to use if the calculated percentage for the current user falls in the provided range.",
+                            "default": "",
+                            "pattern": "^(.*)$"
+                        },
+                        "from": {
+                            "$id": "#/properties/allocation/properties/percentile/items/properties/from",
+                            "type": "number",
+                            "title": "Percentile Allocation From",
+                            "description": "The lower end of the percentage range for which this variant will be used.",
+                            "default": 0
+                        },
+                        "to": {
+                            "$id": "#/properties/allocation/properties/percentile/items/properties/to",
+                            "type": "number",
+                            "title": "Percentile Allocation To",
+                            "description": "The upper end of the percentage range for which this variant will be used.",
+                            "default": 100
+                        }
+                    }
+                }
+            },
+            "seed": {
+                "$id": "#/properties/allocation/properties/seed",
+                "type": "string",
+                "title": "Percentile Allocation Seed",
+                "description": "The value which percentile calculations are based on. Calculated percentages are consistent across features if the user and seed are consistent.",
+                "default": "",
+                "pattern": "^(.*)$"
+            }
+        }
+      },    
+      "telemetry_enabled": {
+        "$id": "#/properties/telemetry_enabled",
+        "type": "boolean",
+        "title": "Feature Telemetry Enabled State",
+        "description": "A flag to enable or disable sending telemetry events.",
+        "default": false
+      },
+      "telemetry_metadata": {
```

Comment:
This is not a desired property.

Reply:
I expect it is. We will insert this dynamically, but if someone hardcodes this in a feature flag, it will be honored.

Reply:
Why would a customer not use tags on the key-value?

Reply:
Discussed with Zhenlan. If we were to have feature flags as a resource at the same level of key-values we would most likely have telemetry configuration on top of existing App Configuration paradigms such as tags and etag. Thus, it is reasonable to include it here not only for feature management client operability but also because it makes sense to exist on the server.

Although since we have multiple properties revolving around telemetry we discussed to switch the schema. Here is an example following the updated schema

{
    "telemetry": {
        "enabled": true,
        "metadata": {
            "foo": "bar"
        }
    }
}